incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/config/MqttKafkaHeaderHelper.java

```diff
@@ -31,54 +28,57 @@
-        kafkaLocalHeaderOctets = stringToOctets(KAFKA_LOCAL_HEADER_NAME);
-        kafkaTimeoutHeaderOctets = stringToOctets(KAFKA_TIMEOUT_HEADER_NAME);
-        kafkaContentTypeHeaderOctets = stringToOctets(KAFKA_CONTENT_TYPE_HEADER_NAME);
-        kafkaFormatHeaderOctets = stringToOctets(KAFKA_FORMAT_HEADER_NAME);
-        kafkaReplyToHeaderOctets = stringToOctets(KAFKA_REPLY_TO_HEADER_NAME);
-        kafkaCorrelationHeaderOctets = stringToOctets(KAFKA_CORRELATION_ID_HEADER_NAME);
-    }
+    private static final String KAFKA_TIMEOUT_HEADER_NAME = "zilla:timeout-ms";
+
+    private static final String KAFKA_CONTENT_TYPE_HEADER_NAME = "zilla:content-type";
+
+    private static final String KAFKA_FORMAT_HEADER_NAME = "zilla:format";
+
+    private static final String KAFKA_REPLY_TO_HEADER_NAME = "zilla:reply-to";
+
+    private static final String KAFKA_CORRELATION_ID_HEADER_NAME = "zilla:correlation-id";
+
+    public final OctetsFW kafkaTopicHeaderName;
+    public final OctetsFW kafkaLocalHeaderName;
+    public final OctetsFW kafkaTimeoutHeaderName;
+    public final OctetsFW kafkaContentTypeHeaderName;
+    public final OctetsFW kafkaFormatHeaderName;
+    public final OctetsFW kafkaReplyToHeaderName;
+    public final OctetsFW kafkaCorrelationHeaderName;
+    private final Map<OctetsFW, Consumer<OctetsFW>> visitors;
+    public final OctetsFW contentTypeRO = new OctetsFW();
+    public final OctetsFW replyToRO = new OctetsFW();
 
-    public List<String> topicNames;
-    public String local;
     public int timeout;
-    public String contentType;
+    public OctetsFW contentType;
     public String format;
-    public String replyTo;
+    public OctetsFW replyTo;
     public OctetsFW correlation;
-    public Map<String, List<String>> userProperties;
+    public IntArrayList userPropertiesOffsets;
+
+    public MqttKafkaHeaderHelper()
+    {
+        kafkaTopicHeaderName = stringToOctets(KAFKA_TOPIC_HEADER_NAME);
+        kafkaLocalHeaderName = stringToOctets(KAFKA_LOCAL_HEADER_NAME);
+        kafkaTimeoutHeaderName = stringToOctets(KAFKA_TIMEOUT_HEADER_NAME);
+        kafkaContentTypeHeaderName = stringToOctets(KAFKA_CONTENT_TYPE_HEADER_NAME);
+        kafkaFormatHeaderName = stringToOctets(KAFKA_FORMAT_HEADER_NAME);
+        kafkaReplyToHeaderName = stringToOctets(KAFKA_REPLY_TO_HEADER_NAME);
+        kafkaCorrelationHeaderName = stringToOctets(KAFKA_CORRELATION_ID_HEADER_NAME);
+
+        visitors = new HashMap<>();
+        visitors.put(kafkaTopicHeaderName, this::skip);
+        visitors.put(kafkaLocalHeaderName, this::skip);
+        visitors.put(kafkaTimeoutHeaderName, this::visitTimeout);
+        visitors.put(kafkaContentTypeHeaderName, this::visitContentType);
+        visitors.put(kafkaFormatHeaderName, this::visitFormat);
+        visitors.put(kafkaReplyToHeaderName, this::visitReplyTo);
+        visitors.put(kafkaCorrelationHeaderName, this::visitCorrelationId);
+    }
 
     private OctetsFW stringToOctets(String string0)
```

Comment:
Please make this method static at the end of the class as it is called from constructor and does not depend on instance state.

Also rename input parameter to avoid `0` and fix parameter formatting to next line.